chall04/vde-dios.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+import sys
+import re
+
+
+def error(e , i):
+    file_name =  "stdin" if i == 0 else sys.argv[i]
+    if (e == 1):
+        print("%s: %s: Bad format" %(sys.argv[0], file_name))
+    elif (e == 2):
+        print("%s: %s: Can't read file" %(sys.argv[0], file_name))
+    elif (e == 3):
+        print("%s: %s: Not enough space in the given shelves" %(sys.argv[0], file_name))
+
+
+# Check format plus sort shelves
+def shelf_parse(shelves):
+    l = shelves.replace("\n", "").split(" ")
+    l.sort(key=len, reverse=True)
+    for c in l:
+        if not c.isdigit():
+            return None
```

Comment:
Why would you sort by len? If you want to somehow give a priority to the largest numbers, this won't work if I pass `000001` as a shelf, the value would be `1` but it would be placed before `400`.  
Something more logical would have been to sort it by value:
```suggestion
    if not "".join(l).isdigit():
        return None
    l = sorted([int(s) for s in l], reverse=True)
```
Also, try using more explicit var names `l` is really short and might get confusing (as you can see in the comprehension I wrote)

Reply:
Totally agree with that. I realized later I was trying to sort strings rather than integers which does not work in the same way